### PR TITLE
DashboardScene: Panel Keybindings and some others 

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Add small util to make it easier to bind and unbind keybindings 
* [x] Create version of withFocusedPanel that works for scene panels 
* [x] Refactor out url generation functions to it's own util file and share view panel url generation with panel menu action
* [x] Edit keybinding 
* [x] Share keybinding
* [x] Toggle legend 

For future PRs
* Time keybindings
* Other keybindings

think I will wait for unit tests, most of the important code (url generation and others) here already has tests and these actions are mostly simple. Think when we add "Toggle all legends" and "toggle all exemplars" we should have some tests for those (and the "toggle legend" on panel level.

Depends on https://github.com/grafana/scenes/pull/403

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        }
+      }
+    }
+  };
+}
+
+export function toggleVizPanelLegend(vizPanel: VizPanel) {
+  const options = vizPanel.state.options;
+  if (hasLegendOptions(options) && typeof options.legend.showLegend === 'boolean') {
+    vizPanel.setState({
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:114]: You can use `vizPanel.onOptionsChange` and just pass legend options.

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        const vizPanel = sceneGraph.findObject(scene, (o) => o.state.key === panelKey);
+
+        if (vizPanel && vizPanel instanceof VizPanel) {
+          fn(vizPanel);
+          return;
+        }
+      }
+    }
+  };
+}
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:109]: Interesting, that's a clever approach!

**Review by dprokop — APPROVED** [review]
Works like expected. Left one nit in the code other than that it's great.

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/scene/keyboardShortcuts.ts]: Pushed an update for this

Comment (torkelo) [reply public/app/features/dashboard-scene/scene/keyboardShortcuts.ts]: thanks

**Pull request merged**